### PR TITLE
TST: Rank - Fix rank entropy and OTSU tests

### LINKED ISSUE #2528: Rank test fails (or too sensitive?) on 32-bit
```

======================================================================
FAIL: skimage.filters.rank.tests.test_rank.test_all
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/venv/local/lib/python2.7/site-packages/skimage/filters/rank/tests/test_rank.py", line 16, in test_all
    check_all()
  File "/venv/local/lib/python2.7/site-packages/skimage/_shared/testing.py", line 232, in inner
    result = func(*args, **kwargs)
  File "/venv/local/lib/python2.7/site-packages/skimage/filters/rank/tests/test_rank.py", line 83, in check_all
    rank.entropy(image, selem))
  File "/venv/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 257, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/venv/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 719, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/venv/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 645, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 1.28%)
 x: array([[ 1.5849625 ,  2.        ,  2.        ,  2.        ,  2.        ,
         2.        ,  2.        ,  2.        ,  2.        ,  2.        ,
         2.        ,  2.        ,  2.        ,  2.        ,  2.        ,...
 y: array([[ 1.5849625 ,  2.        ,  2.        ,  2.        ,  2.        ,
         2.        ,  2.        ,  2.        ,  2.        ,  2.        ,
         2.        ,  2.        ,  2.        ,  2.        ,  2.        ,...
```

### PR DESCRIPTION
**Pull request by hmaarrfk**
Fixes #2528

I think `parametrize` will help in the future, it definitely helped me!

This is in a seperate PR so it can be backported easily.

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/filters/rank/tests/test_rank.py [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 from skimage.filters import rank
+from skimage.filters.rank import __all__ as all_rank_filters
 from skimage._shared._warnings import expected_warnings
-from skimage._shared.testing import test_parallel, xfail, arch32
+from skimage._shared.testing import test_parallel, arch32, parametrize, xfail
+from pytest import param
+
+
+def test_otsu_edge_case():
+    # This is an edge case that causes OTSU to appear to misbehave
```
[the remark this answers is not in the document]
Comment (hmaarrfk) [reply skimage/filters/rank/tests/test_rank.py]: Stefan please see https://github.com/scikit-image/scikit-image/issues/2528#issuecomment-432278657

[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/filters/rank/tests/test_rank.py]: I saw that, but I don't understand: "compiler-dependent" yes, but "rounding error"—if a compiler rounds in error, then we should not return the incorrect result.

[the remark this answers is not in the document]
Comment (hmaarrfk) [reply skimage/filters/rank/tests/test_rank.py]: Yeah sorry, I don't know what the cause really is, I just know both answers are mathematically close to 1E-15. So depending on the specific  circumstances, one of the two values will be chosen. I rephrased my original comment:
```
An error on the order of 1E-15 (floating64 point precision) is causing the algorithm threshold to change between two equivalent values, 41 and 81. This can be caused maybe by:
1. A decision to reorder multiplying or dividing by a constant.
2. A 32bit on 64 bit host emulation error????
3. Some optimization the compiler is doing???
4. A hardware error? doubtful? but maybe
```

**Comment by soupault** [comment]
@meeseeksdev backport to v0.14.x
